ribs/archives/_cvt_archive.py

```diff
@@ -259,13 +264,36 @@ def index_of(self, measures_batch):
         if self._use_kd_tree:
             _, indices = self._centroid_kd_tree.query(measures_batch)
             return indices.astype(np.int32)
-
-        # Brute force distance calculation -- start by taking the difference
-        # between each measure i and all the centroids.
-        distances = np.expand_dims(measures_batch, axis=1) - self.centroids
-
-        # Compute the total squared distance -- no need to compute actual
-        # distance with a sqrt.
-        distances = np.sum(np.square(distances), axis=2)
-
-        return np.argmin(distances, axis=1).astype(np.int32)
+        else:
+            expanded_measures = np.expand_dims(measures_batch, axis=1)
+            # Compute indices chunks at a time
+            if (self._chunk_size is not None and
+                    self._chunk_size < measures_batch.shape[0]):
+                distances = np.sum(
+                    np.square(expanded_measures[0:min(len(measures_batch), self.
+                                                      _chunk_size)] -
+                              self.centroids),
+                    axis=2)
+                indices = [np.argmin(distances, axis=1).astype(np.int32)]
+                for i in range(
```

Comment:
See if you can use `np.array_split` here -- should be cleaner. You'll have to compute the number of chunks for array_split as `np.ceil(len(expanded_measures) / chunk_size)`